docs/guides/testing-keycloak.rst

```diff
@@ -0,0 +1,111 @@
+Testing Keycloak integration
+############################
+
+Keycloak integration can be tested using the related methods provided in
+:class:`sssd_test_framework.roles.keycloak.Keycloak`.
+
+Multhost test configuration requirements
+----------------------------------------
+
+Testing with Keycloak integration requires additional systems configured to
+provide a Keycloak server and the ipa-tuura SCIMv2 **bridge** service:
+
+.. code-block:: yaml
+
+    domains:
+    - id: sssd
+      hosts:
+      - hostname: idm.ipa.test
+        role: ipa
+        ssh:
+          host: HOSTNAME_IPA
+          username: root
+          password: Secret123
+        config:
+          client:
+          ipa_domain: ipa.test
+
+      - hostname: bridge.ipa.test
+        role: client
+        ssh:
+          host: HOSTNAME_BRIDGE
+          username: root
+          password: Secret123
+
+      - hostname: keycloak.ipa.test
+        role: keycloak
+        ssh:
+          host: HOSTNAME_KEYCLOAK
+          username: root
+          password: Secret123
+        config:
+          keycloak:
+            keycloak_server: keycloak.ipa.test
+            keycloak_admin_user: admin
+            keycloak_admin_password: Secret123
+
+It should be noted that at this time, the ipa-tuura bridge service does not
+yet have its own role.  Instead it is defined as a client for remote access if
+necessary and is managed via the Keycloak SCIM storage plugin.
+
+Adding a Keycloak Integration Domain
+------------------------------------
+
+Keycloak integration requires the Keycloak server to add an Integration Domain
+that will point to the identity provider being tested (IPA, AD, LDAP, etc).
+This is accomplished using the method
+:meth:`sssd_test_framework.roles.keycloak.KeycloakDomain.add`.   After Adding
+the Integration Domain, you can add users to keycloak and they should be synced
+to the identity provider.  At that point, you should be able to use them like
```

Comment:
Double space before At.

Reply:
fixing.

Reply:
This was also fixed then moved elsewhere.